src/uint/secret_signed.rs

```diff
@@ -97,21 +90,19 @@ where
     /// Creates a [`SignedSecret`] from an unsigned value, treating it as if it encodes the sign as two's complement.
     ///
     /// Returns `None` if the requested bound is too large, or if `abs(value)` is actually larger than the bound.
-    pub fn new_from_unsigned(value: Secret<T>, bound: u32) -> Option<Self> {
-        let is_negative = Choice::from(value.expose_secret().bit_vartime(T::BITS - 1) as u8);
+    pub fn new_from_unsigned(value: Secret<T>, bound: u32) -> CtOption<Self> {
+        let is_negative = value.expose_secret().bit(T::BITS - 1);
         let abs = Secret::<T>::conditional_select(&value, &value.wrapping_neg(), is_negative);
         // Reserving one bit as the sign bit (MSB)
-        if bound >= T::BITS || abs.expose_secret().bits() > bound {
-            return None;
-        }
-        Some(Self::new_from_unsigned_unchecked(value, bound))
+        let in_bound = Choice::from((bound < T::BITS && abs.expose_secret().bits() <= bound) as u8);
```

Comment:
Need to document that it's variable-time w.r.t. `bound`

Reply:
Not sure what you mean here: after this PR it is not variable time in the bound, so not sure what to document!